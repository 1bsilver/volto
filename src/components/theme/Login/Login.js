/**
 * Login container.
 * @module components/theme/Login/Login
 */

import React, { PropTypes, Component } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';

import { login } from '../../../actions';

@connect(
  state => ({
    loaded: state.userSession.login.loaded,
    error: state.userSession.login.error,
    token: state.userSession.token,
  }),
  dispatch => bindActionCreators({ login }, dispatch),
)
/**
 * Login container class.
 * @class Login
 * @extends Component
 */
export default class Login extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    login: PropTypes.func.isRequired,
    loaded: PropTypes.bool.isRequired,
    error: PropTypes.object,
    token: PropTypes.string,
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      browserHistory.push('/');
    }
  }

  /**
   * On login handler
   * @method onLogin
   * @returns {undefined}
   */
  onLogin() {
    this.props.login(document.getElementById('login').value,
                     document.getElementById('password').value);
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div id="page-login">
        <Helmet title="Login" />
        <div className="container">
          {this.props.error &&
            <div className="portalMessage error">
              <strong>Error</strong>
               {this.props.error.message}
            </div>
          }
          <div className="field">
            <label forHtml="login" className="horizontal">
              Login Name <span className="required horizontal" title="Required">&nbsp;</span>
            </label>
              
            <input id="login" name="login" className="text-widget required textline-field" type="text" />
          </div>

          <div className="field">
            <label forHtml="password" className="horizontal">
              Password <span className="required horizontal" title="Required">&nbsp;</span>
            </label>
              
            <input id="password" name="password" className="text-widget required textline-field" type="password" />
          </div>

          <div className="formControls">
            <button className="context" onClick={::this.onLogin}>Log In</button>
          </div>
        </div>
      </div>
    );
  }
}
