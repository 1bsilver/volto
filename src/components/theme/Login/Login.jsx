/**
 * Login container.
 * @module components/theme/Login/Login
 */

import React, { PropTypes, Component } from 'react';
import Helmet from 'react-helmet';
import { asyncConnect } from 'redux-connect';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';
import { isEmpty } from 'lodash';

import { login } from '../../../actions';

@asyncConnect(
  [
    {
      key: 'userSession',
      promise: ({ store: { dispatch, getState } }) => {
        const form = getState().form;
        if (!isEmpty(form)) {
          return dispatch(login(form.login, form.password));
        }
        return Promise.resolve({});
      },
    },
  ],
)
@connect(
  state => ({
    error: state.userSession.login.error,
    token: state.userSession.token,
  }),
  dispatch => bindActionCreators({ login }, dispatch),
)
/**
 * Login container class.
 * @class Login
 * @extends Component
 */
export default class Login extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    login: PropTypes.func.isRequired,
    error: PropTypes.shape({
      message: PropTypes.string,
    }),
    token: PropTypes.string, // eslint-disable-line react/no-unused-prop-types
  }

  /**
   * Default properties.
   * @property {Object} defaultProps Default properties.
   * @static
   */
  static defaultProps = {
    error: null,
    token: null,
  }

  /**
   * Constructor
   * @method constructor
   * @param {Object} props Component properties
   * @constructs WysiwygEditor
   */
  constructor(props) {
    super(props);
    this.onLogin = this.onLogin.bind(this);
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      browserHistory.push('/');
    }
  }

  /**
   * On login handler
   * @method onLogin
   * @param {Object} event Event object.
   * @returns {undefined}
   */
  onLogin(event) {
    this.props.login(document.getElementById('login').value,
                     document.getElementById('password').value);
    event.preventDefault();
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    return (
      <div id="page-login">
        <Helmet title="Login" />
        <div className="container">
          {this.props.error &&
            <div className="portalMessage error">
              <strong>Error</strong>
              {this.props.error.message}
            </div>
          }
          <form method="post">
            <div className="field">
              <label htmlFor="login" className="horizontal">
                Login Name <span className="required horizontal" title="Required">&nbsp;</span>
              </label>
              <input id="login" name="login" className="text-widget required textline-field" type="text" />
            </div>

            <div className="field">
              <label htmlFor="password" className="horizontal">
                Password <span className="required horizontal" title="Required">&nbsp;</span>
              </label>
              <input id="password" name="password" className="text-widget required textline-field" type="password" />
            </div>

            <div className="formControls">
              <button className="context" onClick={this.onLogin}>Log In</button>
            </div>
          </form>
        </div>
      </div>
    );
  }
}
