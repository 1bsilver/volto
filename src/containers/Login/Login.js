/**
 * Login container.
 * @module components/
 */

import React, { PropTypes, Component } from 'react';
import Helmet from 'react-helmet';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { browserHistory } from 'react-router';

import { login } from 'actions';

@connect(
  state => ({
    loaded: state.login.loaded,
    error: state.login.error,
    token: state.login.token,
  }),
  dispatch => bindActionCreators({ login }, dispatch),
)
/**
 * Login container class.
 * @class Login
 * @extends Component
 */
export default class Login extends Component {

  /**
   * Property types.
   * @property {Object} propTypes Property types.
   * @static
   */
  static propTypes = {
    login: PropTypes.func.isRequired,
    loaded: PropTypes.bool.isRequired,
    error: PropTypes.object,
    token: PropTypes.string,
  }

  /**
   * Component will receive props
   * @method componentWillReceiveProps
   * @param {Object} nextProps Next properties
   * @returns {undefined}
   */
  componentWillReceiveProps(nextProps) {
    if (nextProps.token) {
      localStorage.setItem('auth_token', nextProps.token);
      browserHistory.push('/');
    }
  }

  /**
   * On login handler
   * @method onLogin
   * @returns {undefined}
   */
  onLogin() {
    this.props.login(document.getElementById('login').value,
                     document.getElementById('password').value);
  }

  /**
   * Render method.
   * @method render
   * @returns {string} Markup for the component.
   */
  render() {
    console.log(this.props.error);
    return (
      <div id="page-login">
        <Helmet title="Login" />
        <div className="container">
          {this.props.error &&
            <div className="portalMessage error">
              <strong>Error</strong>
               {this.props.error.message}
            </div>
          }
          <div className="field">
            <label forHtml="login" className="horizontal">
              Login Name <span className="required horizontal" title="Required">&nbsp;</span>
            </label>
              
            <input id="login" name="login" className="text-widget required textline-field" type="text" />
          </div>

          <div className="field">
            <label forHtml="password" className="horizontal">
              Password <span className="required horizontal" title="Required">&nbsp;</span>
            </label>
              
            <input id="password" name="password" className="text-widget required textline-field" type="password" />
          </div>

          <div className="formControls">
            <button className="context" onClick={::this.onLogin}>Log In</button>
          </div>
        </div>
      </div>
    );
  }
}
