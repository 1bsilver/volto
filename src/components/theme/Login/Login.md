Login example source:

```jsx_ noeditor
const { Provider } = require('react-intl-redux');
const configureStore = require('redux-mock-store').default;
import {BrowserRouter} from 'react-router-dom';
import { LoginComponent } from './Login';
const store = configureStore()({
  userSession: {
    login: {},
  },
  intl: {
    locale: 'en',
    messages: {},
  },
});

<div className={'rsg--pre-42'}>
  <Provider store={store}>
    <BrowserRouter>
      <LoginComponent
        error={(message = '')}
        token=""
        login={form => null}
        location={{}}
      />
    </BrowserRouter>
  </Provider>
</div>;
```
