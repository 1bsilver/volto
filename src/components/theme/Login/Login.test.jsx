import React from 'react';
import renderer from 'react-test-renderer';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

import { LoginComponent as Login } from './Login';

const mockStore = configureStore();

describe('Login', () => {
  it('renders a login component', () => {
    const store = mockStore({
      userSession: {
        login: {},
      },
    });
    const component = renderer.create(
      <Provider store={store}>
        <Login />
      </Provider>,
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
