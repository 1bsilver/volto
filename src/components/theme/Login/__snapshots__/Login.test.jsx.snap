// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders a login component 1`] = `
<div
  id="page-login"
>
  <div
    className="container"
  >
    <form
      className="ui form"
      method="post"
      onSubmit={[Function]}
    >
      <div
        className="ui top attached segment"
      >
        <div
          className="field"
        >
          <label
            htmlFor="login"
          >
            Login Name
          </label>
          <div
            className="ui input"
          >
            <input
              id="login"
              name="login"
              type="text"
            />
          </div>
        </div>
        <div
          className="field"
        >
          <label
            htmlFor="password"
          >
            Password
          </label>
          <div
            className="ui input"
          >
            <input
              id="password"
              name="password"
              type="password"
            />
          </div>
        </div>
      </div>
      <div
        className="ui bottom attached segment"
      >
        <button
          className="ui primary button"
          onClick={[Function]}
          tabIndex={undefined}
          type="submit"
        >
          Log In
        </button>
      </div>
    </form>
  </div>
</div>
`;
